flik/linesearch/conditions.py

```diff
@@ -1 +1,47 @@
 """Conditions for the line search algorithm."""
+def soft_wolfe(gradf, valf, alpha, direction, const2=0.9):
+    """Curvature condition
+
+    Parameters
+    ----------
+    gradf: function
+        Gradient of objective nonlinear function
+    valf: np.ndarray(N,)
+        Function value
+    alpha: float
+        Step length
+    direction: np.ndarray(N,)
+        Descent direction
+    const2: float
+        Constant C2
+
+    Raises
+    ------
+    ValueError
+        If C2 is not between 0 and 1
+    TypeError
+        If gradient function if not a function
+
+    Returns
+    -------
+    Boolean
+        True if curvature condition is satisfied
+    """
+    # Check parameters
+    if not isinstance(valf, (np.ndarray, float)):
+        raise TypeError('Variable should be an array of float')
+    if not isinstance(direction, (np.ndarray, float)):
+       raise TypeError('Descent direction should be an array of float')
+    if not isinstance(alpha, float):
+        raise TypeError("Alpha should be a float")
+    if xk.size != sk.size:
+        raise ValueError('Size of direction and variable should be same')
+    if not 0 < const2 < 1:
```

Comment:
we should first check that `const2` is a float. If it is a numpy array, the code will raise an error here (because you cannot use `not` operator with a numpy array).